map/gps_tracker.cpp

```diff
@@ -111,3 +116,13 @@ void GpsTracker::OnLocationUpdated(location::GpsInfo const & info)
     return;
   m_track.AddPoint(info);
 }
+
+bool GpsTracker::IsRecentTrackCollectionInitialized() const
```

Comment:
When/why is it needed? Is there some polling from UI until the collection is loaded?

Reply:
@kavikhalique can you please help me with the answer?

Reply:
The check internally does this:
```
bool GpsTrack::IsTrackerInitialized() const
{
  std::lock_guard lg(m_dataGuard);
  return m_collection != nullptr;
}
```

Collection is initialized when the callback from the UI has been set. @kavikhalique can you please clarify your use-case in the UI, why this check is needed? Even if collection is not _yet_ initialized, but will be initialized later (after setting a callback), what will break in your code?

Reply:
@biodranik I removed the `IsTrackerInitialized` and `IsRecentTrackCollectionInitialized` and use the IsEmpty instead of in the `SaveTrackRecording`, because this method checks that m_collection exists:
```
std::unique_ptr<GpsTrackCollection> m_collection

bool GpsTracker::IsEmpty() const
{
  return m_track.IsEmpty();
}

bool GpsTrack::IsEmpty() const
{
  if (!m_collection)
    return true;
  return m_collection->IsEmpty();
}
```